src/indexer.c

```diff
@@ -291,6 +294,53 @@ static void doAssignIds(RSAddDocumentCtx *cur, RedisSearchCtx *ctx) {
   }
 }
 
+static void indexBulkFields(RSAddDocumentCtx *aCtx, RedisSearchCtx *sctx) {
+  // Traverse all fields, seeing if there may be something which can be written!
+  IndexBulkData bData[SPEC_MAX_FIELDS] = {{0}};
+  IndexBulkData *activeBulks[SPEC_MAX_FIELDS];
```

Comment:
maybe its better to use the arr object in util for those two

Reply:
Why use a dynamic array when I can use a static one?

Reply:
there will be no need to save the size and increasing it, just a suggestion ...